include/nana/gui/widgets/listbox.hpp

```diff
@@ -461,13 +461,25 @@ namespace nana
 		}
 	}//end namespace drawerbase
 
-	struct arg_listbox
+    struct arg_listbox
+        : public event_arg
+    {
+        mutable drawerbase::listbox::item_proxy item;
+        bool    selected;
+
+        arg_listbox(const drawerbase::listbox::item_proxy&, bool selected);
+    };
+	struct arg_item
 		: public event_arg
 	{
-		mutable drawerbase::listbox::item_proxy item;
-		bool	selected;
-
-		arg_listbox(const drawerbase::listbox::item_proxy&, bool selected);
+        drawerbase::listbox::index_pair item;
+        drawerbase::listbox::size_type column;
+        void block_category_change() const;
+        bool category_change_blocked() const;
```

Comment:
Thanks for your pull request, I think it would be much better if this event only occurs when double click on a category to determine whether to change the status of expansion of a category. Because block_category_change() and category_change_blocked() may confuse us when clicking on an item(not category).

By using general double click event, it is able to handle the double click of an item.
```C++
list.events().dbl_click([&list](const arg_mouse& arg)
{
	auto s = list.selected();
	if (s.empty())
	{
		//auto col = list.column(arg.pos); //Add a method to get the column index with a specified coordinate
	}
});
```